node/impl/ipc/ipc.go

```diff
@@ -93,3 +97,33 @@ func (a *IpcAPI) IpcAddSubnetActor(ctx context.Context, wallet address.Address,
 	}
 	return r.IDAddress, nil
 }
+
+func (a *IpcAPI) IpcReadGatewayState(ctx context.Context, actor address.Address, tsk types.TipSetKey) (*gateway.State, error) {
+	st := &gateway.State{}
+	if err := a.readActorState(ctx, actor, tsk, &gateway.State{}); err != nil {
+		return nil, xerrors.Errorf("error getting gateway actor from StateStore: %w")
+	}
+	return st, nil
+}
+
+func (a *IpcAPI) IpcReadSubnetActorState(ctx context.Context, actor address.Address, tsk types.TipSetKey) (*subnetactor.State, error) {
+	st := &subnetactor.State{}
+	if err := a.readActorState(ctx, actor, tsk, st); err != nil {
+		return nil, xerrors.Errorf("error getting subnet actor from StateStore: %w")
+	}
+	return st, nil
+}
+
+func (a *IpcAPI) readActorState(ctx context.Context, actor address.Address, tsk types.TipSetKey, stateType cbg.CBORMarshaler) error {
```

Comment:
This method might deserve a small comment explaining it.

Reply:
Agree. Fixed.